spring-batch-infrastructure/src/main/java/org/springframework/batch/item/database/support/DefaultDataFieldMaxValueIncrementerFactory.java

```diff
@@ -74,50 +79,46 @@ public DefaultDataFieldMaxValueIncrementerFactory(DataSource dataSource) {
 		this.dataSource = dataSource;
 	}
 
-    @Override
+	@Override
 	public DataFieldMaxValueIncrementer getIncrementer(String incrementerType, String incrementerName) {
 		DatabaseType databaseType = DatabaseType.valueOf(incrementerType.toUpperCase());
 
-		if (databaseType == DB2) {
+		switch (databaseType) {
+		// DB2 and DB2AS400 use the same syntax for retrieval of next value from a sequence
+		case DB2:
+		case DB2AS400:
 			return new DB2SequenceMaxValueIncrementer(dataSource, incrementerName);
-		}
-		else if (databaseType == DB2ZOS) {
+		case DB2ZOS:
 			return new DB2MainframeSequenceMaxValueIncrementer(dataSource, incrementerName);
-		}
-		else if (databaseType == DERBY) {
+		case DERBY:
 			return new DerbyMaxValueIncrementer(dataSource, incrementerName, incrementerColumnName);
-		}
-		else if (databaseType == HSQL) {
-			return new HsqlMaxValueIncrementer(dataSource, incrementerName, incrementerColumnName);
-		}
-		else if (databaseType == H2) {
+		case H2:
 			return new H2SequenceMaxValueIncrementer(dataSource, incrementerName);
-		}
-		else if (databaseType == MYSQL) {
+		case HSQL:
+			return new HsqlMaxValueIncrementer(dataSource, incrementerName, incrementerColumnName);
+		case MYSQL:
 			return new MySQLMaxValueIncrementer(dataSource, incrementerName, incrementerColumnName);
-		}
-		else if (databaseType == ORACLE) {
+		case ORACLE:
 			return new OracleSequenceMaxValueIncrementer(dataSource, incrementerName);
-		}
-		else if (databaseType == POSTGRES) {
+		case POSTGRES:
 			return new PostgreSQLSequenceMaxValueIncrementer(dataSource, incrementerName);
-		}
-		else if (databaseType == SQLITE) {
+		case SQLITE:
 			return new SqliteMaxValueIncrementer(dataSource, incrementerName, incrementerColumnName);
-		}
-		else if (databaseType == SQLSERVER) {
+		case SQLSERVER:
 			return new SqlServerMaxValueIncrementer(dataSource, incrementerName, incrementerColumnName);
-		}
-		else if (databaseType == SYBASE) {
+		case SYBASE:
 			return new SybaseMaxValueIncrementer(dataSource, incrementerName, incrementerColumnName);
-		}
-		throw new IllegalArgumentException("databaseType argument was not on the approved list");
 
+		// DB2VSE was also missing in previous implementation
+		case DB2VSE:
```

Comment:
I'm using very strict compiler settings which cause a warning on an incomplete switch over an enum. I like making things explicitly visible. But it's not actually necessary, so feel free to drop the case. ;-)